lock-api-objects/src/main/java/com/palantir/lock/v2/ExceptionHandlingRunner.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.v2;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+import com.palantir.logsafe.exceptions.SafeRuntimeException;
+
+public final class ExceptionHandlingRunner implements AutoCloseable {
```

Comment:
This class is copied from bottom of SnapshotTransactionManager - probably want to merge the two (if I can find a place where it makes sense to live in the project hierarchy).